types.go

```diff
@@ -14,6 +15,20 @@ var (
 	defaultSum   [sumSize]byte
 )
 
+// MerkleRoot is a type alias for a byte slice returned from the Root method
+type MerkleRoot []byte
+
+// Sum returns the uint64 sum of the merkle root, if sumTrie is true
+// otherwise it returns 0, as it would result in undefined behaviour.
+func (r MerkleRoot) Sum(sumTrie bool) uint64 {
```

Comment:
@h5law I approved it but got an idea. What if we check the length of `r` and just panic if it's not long enough?